### PR TITLE
[FIX] (*_)product_matrix: fixes and tests

### PR DESCRIPTION
**Pull request by Feyensv**

### CODE REVIEW

**Comment by Feyensv** [comment]
@ged-odoo Only two conditions needs to be added in `relational_fields.js` to support a `field_changed` event without operations from a relational child widget (Cf. last commit).  I'll add another test for the matrix, but I don't know how to mock a widget triggering this kind of event in a test for the `web` module :thinking:.

Is it needed or can it be merged only with a test in matrix verifying this case ?

File: addons/product_matrix/static/src/js/section_and_note_widget.js [hunk 1]
```diff
@@ -65,15 +65,24 @@ SectionAndNoteFieldOne2Many.include({
...
                 grid_product_tmpl_id: {id: productTemplateId}
             },
             viewType: 'form',
             onSuccess: function (result) {
-                // why the hell do we receive a table of 8 classes here? :o
-                self._openMatrixConfigurator(result[0].recordData.grid, productTemplateId, editedCellAttributes);
+                // result = list of widgets
+                // find the one of the grid field
+                // and get the grid information from its recordData.
+                var gridInfo = "";
```
Comment (ged-odoo) [inline addons/product_matrix/static/src/js/section_and_note_widget.js]: You can replace the next few lines with this I think:
`var gridInfo = result.find(r => r.recordData.grid);`

**Comment by ged-odoo** [comment]
good work. It's fine. I just made a small comment, and then it can be merged. good job, it's better now, and the test will help us a lot

File: addons/product_matrix/static/src/js/section_and_note_widget.js (hunk 1 above)
Comment (Feyensv) [reply addons/product_matrix/static/src/js/section_and_note_widget.js]: Can we use arrow functions in 13.0 ? (or should I target master for my fiximps?)

Comment (ged-odoo) [reply addons/product_matrix/static/src/js/section_and_note_widget.js]: you can use arrow functions (and all ES6 features)

Comment (Feyensv) [reply addons/product_matrix/static/src/js/section_and_note_widget.js]: Can I remove IE11 from supported browsers in the 13.0 doc then?

Comment (ged-odoo) [reply addons/product_matrix/static/src/js/section_and_note_widget.js]: that would actually be wise... We should actually have done that before

Comment (Feyensv) [reply addons/product_matrix/static/src/js/section_and_note_widget.js]: :arrow_right: https://github.com/odoo/odoo/pull/40602

File: addons/product_matrix/tests/common.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+class TestMatrixCommon(common.HttpCase):
+
+    def setUp(self):
+        super(TestMatrixCommon, self).setUp()
+        # group_product_variant: use the product configurator
+        # group_sale_pricelist: display the pricelist to determine when
+        # it is changed after choosing the partner
+        self.env.ref('base.user_admin').write({
+            'groups_id': [
+                (4, self.env.ref('product.group_product_variant').id),
```
Comment (Feyensv) [inline addons/product_matrix/tests/common.py]: Todo, verify those groups are needed.

**Comment by dbo-odoo** [comment]
@robodoo r+
lepers

**Comment by dbo-odoo** [comment]
@robodoo rebase-ff

**Pull request closed without merging**